app/views/membership.html

```diff
@@ -0,0 +1,292 @@
+                    flex row tablet="column" mobile="column">
+                    <div
+                      class="col-roles"
+                      row tablet="column" flex wrap axis="start">
+                      <action-chip
+                        ng-repeat="role in subject.roles track by $index"
+                        key="role.name"
+                        key-help="role.description || 'There is no additional information about this role'"
+                        show-action="mode.edit"
+                        action="confirmRemove(subject.name, subjectKind.name, role.name)"
+                        action-title="remove role {{role}} from {{subject.name}}"></action-chip>
+                    </div>
+                    <div
+                      class="col-add-role">
+                      <div ng-show="mode.edit" row>
+                        <ui-select
+                          ng-if="filteredRoles.length"
+                          ng-model="subject.newBinding.role"
+                          theme="bootstrap"
+                          search-enabled="true"
+                          title="Select a new role for {{subjectKind.name}}"
+                          class="select-role"
+                          flex>
+                          <ui-select-match
+                            placeholder="Select a role">
+                            <span ng-bind="subject.newBinding.role"></span>
+                          </ui-select-match>
+                          <ui-select-choices
+                            repeat="role.metadata.name as role in filteredRoles | filter: $select.search">
+                            <div>{{role.metadata.name}}</div>
+                            <div ng-if="role | annotation : 'description'">
+                              <small>{{role | annotation : 'description'}}</small>
+                            </div>
+                          </ui-select-choices>
+                        </ui-select>
+                        <a
+                          href=""
+                          ng-disabled="disableAddForm"
+                          ng-click="addRoleTo(subject.name, subjectKind.name, subject.newBinding.role)"
+                          class="btn btn-default add-role-to">
+                          Add
+                        </a>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+
+                <div
+                  ng-if="mode.edit"
+                  class="item-row form-new-role" row>
+                  <div class="col-name" column>
+                    <label
+                      for="new-{{subjectKind.name}}-name"
+                      class="sr-only">
+                      Name
+                    </label>
+                    <input
+                      type="text"
+                      class="form-control input-name"
+                      id="new-{{subjectKind.name}}-name"
```

Comment:
largely because the attributes for and id are picked up by screen readers, you should wait until the content is accurate before they are rendered into the DOM